### ISSUE TITLE
Issues about 'quast-download-busco'

### ISSUE THREAD

**Issue by Carris0819** [ablab/quast#257]
**Hello genius people, I am trying to download busco and quast in my mamba environment, but I have found that there are few error messgaes came through after the last command line of 'quast-download-busco'.**

This is how the error messgaes looks like: 
ERROR! Failed downloading bacteria database (url: https://busco-archive.ezlab.org/v3/datasets/bacteria_odb9.tar.gz), QUAST functionality will be limited! Exception caught: HTTP Error 404: Not Found
You can try to download the file manually, place it in /data/home/carris/miniforge3/envs/sequence_analysis/lib/python3.10/site-packages/quast_libs/busco/bacteria.tar.gz and restart QUAST
WARNING: Failed to download bacteria database from https://busco-archive.ezlab.org/v3/datasets/bacteria_odb9.tar.gz and unpack it into /data/home/carris/miniforge3/envs/sequence_analysis/lib/python3.10/site-packages/quast_libs/busco
  Downloading BUSCO database...
Downloading eukaryota database (file: eukaryota.tar.gz)...

**Does anyone know how to solve this problem? Thank you so much for your help!**

**Comment by alexeigurevich** [comment]
Hello,
It seems that old BUSCO databases are (currently?) unavailable from the [https://busco-archive.ezlab.org](https://busco-archive.ezlab.org/v3/datasets/bacteria_odb9.tar.gz) website, so Quast cannot download and configure them. We plan to switch to the latest BUSCO (v.5) in future Quast releases but the work on Quast is not active nowadays (due to other projects for core developers), so I cannot promise you realistic estimates. 
Note that Quast can work without BUSCO, just a few corresponding quality metrics will be missing. At the same time, you can obtain the latest BUSCO stats using their tool directly: https://busco.ezlab.org/

**Comment by mkweskin** [comment]
The busco v3 website is available on the [Wayback Machine](https://web.archive.org): https://web.archive.org/web/20230606120407/https://busco-archive.ezlab.org/v3/

Surprisingly, the v9 DBs are also archived and even downloadable with wget.

https://web.archive.org/web/20230330151146/https://busco-archive.ezlab.org/v3/datasets/eukaryota_odb9.tar.gz

**Comment by pdimens** [comment]
Has there been any movement on this? The new paths for the most updated databases are:
```
https://busco-data.ezlab.org/v5/data/lineages/
```

**Comment by pmenzel** [comment]
I tried putting the file `https://busco-data.ezlab.org/v5/data/lineages/bacteria_odb10.2024-01-08.tar.gz` into the expected location for Quast (when installed via conda this would be: `$CONDA_PREFIX/lib/pythonX.X/site-packages/quast_libs/busco/bacteria.tar.gz`), and it does not try to download the file, but fails to use it with this message
```
****************** Start a BUSCO 3.0.2 analysis, current time: 11/18/2024 15:53:35 ******************
Configuration loaded from /home/ptr/temp/quast/busco_stats/config.ini
Init tools...
Check dependencies...
The dataset you provided lacks elements in /opt/conda/miniconda3/envs/xxxxxxx/lib/python3.8/site-packages/quast_libs/busco/bacteria/prfl
BUSCO analysis failed !
Check the logs, read the user guide, if you still need technical support, then please contact mailto:support@orthodb.org
```

My guess is, that the older version of BUSCO in Quast cannot use the newer odb10 files.

**Comment by alexeigurevich** [comment]
Yes indeed, we will update BUSCO to the newest version (and databases) in the next release. The new version is also working on macOS, so the long-standing QUAST statement "BUSCO (Linux only)" will also be gone soon.

**Comment by eggrandio** [comment]
Is there any way of getting this "next release" with updated BUSCO ?

**Comment by micro-hg-cpg** [comment]
> Yes indeed, we will update BUSCO to the newest version (and databases) in the next release. The new version is also working on macOS, so the long-standing QUAST statement "BUSCO (Linux only)" will also be gone soon.

Hello, I saw that said this in Nov 2024 and the same month a new release came out but Im currently facing the same problem. Was is added to the last version available?? Is there a way to work around it?

**Comment by alexeigurevich** [comment]
Hello! No, sorry — this has not been implemented or released yet. Currently, QUAST development is progressing rather slowly due to limited resources (i.e., too many other ongoing projects). However, it’s not completely stalled either.

Once we implement the new BUSCO interface, I’ll link this issue in the relevant commits so the progress is clear. As usual, I will also close the issue when the new release supporting this feature becomes available.

**Comment by ShevchenkoAlla** [comment]
Hello, all!
Will this issue be fixed in close future?

I also can't get db for busco.


wget https://busco-archive.ezlab.org/v3/datasets/bacteria_odb9.tar.gz
--2025-12-08 16:54:46--  https://busco-archive.ezlab.org/v3/datasets/bacteria_odb9.tar.gz
Resolving busco-archive.ezlab.org (busco-archive.ezlab.org)... 129.194.190.40
Connecting to busco-archive.ezlab.org (busco-archive.ezlab.org)|129.194.190.40|:443... failed: Connection timed out.
Retrying.

--2025-12-08 16:57:05--  (try: 2)  https://busco-archive.ezlab.org/v3/datasets/bacteria_odb9.tar.gz
Connecting to busco-archive.ezlab.org (busco-archive.ezlab.org)|129.194.190.40|:443... failed: Connection timed out.
Retrying.

--2025-12-08 16:59:17--  (try: 3)  https://busco-archive.ezlab.org/v3/datasets/bacteria_odb9.tar.gz
Connecting to busco-archive.ezlab.org (busco-archive.ezlab.org)|129.194.190.40|:443... failed: Connection timed out.
Retrying.

Thanks.

**Comment by Naugron** [comment]
If anyone needs the old Busco databases, I’ve managed to recover them from my old computer, where I had quast-5.0.0 installed.
https://drive.google.com/drive/folders/159sVFjgrV924e6HUFDltcJzPcia5NWoe
You need to place them in the busco folder where quast is installed. For example, on my system the path is /home/user/miniforge3/envs/quast/lib/python3.13/site-packages/quast_libs/busco/
You’ll need to rename them to bacteria.tar.gz, eukaryota.tar.gz and fungi.tar.gz respectively, then rerun the quast-download-busco and everything should install. At least that’s how it worked for me, but I haven’t checked yet to see if they work.
